src/v/archival/probe.cc

```diff
@@ -105,6 +105,20 @@ void ntp_level_probe::setup_public_metrics(const model::ntp& ntp) {
            "This may grow due to retention or non compacted segments being "
            "replaced with their compacted equivalent."),
          labels)
+         .aggregate(aggregate_labels),
+       sm::make_gauge(
+         "segments",
+         [this] { return _segments_in_manifest; },
+         sm::description(
+           "Total number of accounted segments in the cloud for the topic"),
+         labels)
+         .aggregate(aggregate_labels),
+       sm::make_gauge(
+         "deletion_backlog_size",
```

Comment:
Let's keep a noun in here so that we know in the name what the backlog is a backlog _of_, and to be a bit more consistent with the naming of `segments` above.

Maybe `segments_pending_deletion`?

Reply:
Done.